core/src/main/scala/chisel3/Data.scala

```diff
@@ -475,13 +490,6 @@ abstract class Data extends HasId with NamedComponent with SourceInfoDoc {
   private var _specifiedDirection:         SpecifiedDirection = SpecifiedDirection.Unspecified
   private[chisel3] def specifiedDirection: SpecifiedDirection = _specifiedDirection
   private[chisel3] def specifiedDirection_=(direction: SpecifiedDirection) = {
-    if (_specifiedDirection != SpecifiedDirection.Unspecified) {
-      this match {
-        // Anything flies in compatibility mode
-        case t: Record if !t.compileOptions.dontAssumeDirectionality =>
-        case _ => throw RebindingException(s"Attempted reassignment of user-specified direction to $this")
-      }
-    }
```

Comment:
Why did you delete this? I think this should not be deleted.

Reply:
We discussed this offline, but basically we're not sure that this actually can be hit by user code and it prevents the optimization we're doing here. It's possible that removing this will cause someone to get a less great error but we could not come up with any cases where this is anything but an internal error.